pull_request_template.md

```diff
@@ -0,0 +1,41 @@
+:hand:&nbsp;&nbsp;&nbsp;_**Wait!**_
+
+Let's make sure your pull request is ready. Ask yourself...
+
+:white_check_mark:&nbsp;&nbsp;&nbsp;"Did I do a good job testing my own code?"
+
+:white_check_mark:&nbsp;&nbsp;&nbsp;"Is my code easy to read and understand?"
+
+:white_check_mark:&nbsp;&nbsp;&nbsp;"Have I written unit tests? Do the unit tests fully cover my code?"
+
+:white_check_mark:&nbsp;&nbsp;&nbsp;"Have I tested the repository's behavior to make sure my code didn't break anything?"
+
+:negative_squared_cross_mark:&nbsp;&nbsp;&nbsp;"Did I include unrelated work that could be in its own pull request?"
+
+---
+
+# Overview
+Talk about the intent and scope of the original issue. Go over the general changes that have been made between this branch and the branch the pull request will merge into. Displaying screenshots can help!
+
+Add links to any documentation or issues that can help give more background information. Be sure to link GitHub issues with [closing keywords](https://docs.github.com/articles/closing-issues-using-keywords), if possible.
+
+> This pull request resolves/closes/fixes #_issue_.
+
+## Anything else?
+_Optional_. If appropriate, provide additional notes the reviewers should be made aware of. They can be about any "gotchas" or even changes that are expected to be made in the future, but are not applied to this pull request.
+
+## Testing
+List instructions on how to test the pull request. Some examples:
+
+- Run the tests to make sure they pass (`docker-compose run --rm web npm run test`).
```

Comment:
For the backend tests, the command is: `docker-compose run --rm web bundle exec rspec`

This should probably mention that the testing action should pass after each push. (I'll set that up today.)

Reply:
Are there any repositories where we do testing other than `rspec`? I set up Mocha/Chai for the Pride rewrite, which isn't live yet, and may never be. I hope unit testing will become standard for current/future projects. Do you think it would make more sense to replace that line with:
```
- Run the tests to make sure they pass (`docker-compose run --rm web bundle exec rspec`).
 ```
```
- Run the tests to make sure they pass (`docker-compose run --rm web bundle exec rspec`/`docker-compose run --rm web npm run test`).
```
or
```
- Run the tests to make sure they pass (`docker-compose run --rm web bundle exec rspec`).
  - Break the new/updated unit tests to make sure they're working properly.
- Run the tests to make sure they pass (`docker-compose run --rm web npm run test`).
  - Break the new/updated unit tests to make sure they're working properly.
``

The template does a lot of explaining, but I wouldn't mind having some examples within it where people can just delete/modify to help streamline getting the PR up.

Reply:
For backend ruby tests I've only seen rspec. It's possible there's some minitest, but I haven't encountered any yet that have been maintained. 

That said, there should be tests for perl and php and javascript and whatever appropriate language is being developed, so what about expecting that the project README has instructions on the automated testing procedure for the project?

Reply:
That is a good idea. Maybe in the future we can work on creating a `README.md` template for repositories.

I'm trying to think of most common scenarios within LIT for pull request descriptions, whether the code relates to back-end or front-end. After skimming a few dozen PRs under `mlibrary`, most of them are pretty sparse, so I don't have much to go off of. 🙃

Reply:
Yep. :upside_down_face: 

I do think `docker-compose run --rm web bundle exec rspec` and `docker-compose run --rm web npm run tests` will cover a lot.